bundles/org.openhab.binding.telegram/src/main/java/org/openhab/binding/telegram/bot/TelegramActions.java

```diff
@@ -341,6 +341,15 @@ public boolean sendTelegramPhoto(@ActionInput(name = "chatId") @Nullable Long ch
                 // Load image from provided base64 image
                 logger.debug("Photo base64 provided; converting to binary.");
                 try {
+                    if (photoURL.startsWith("data:")) // support data URI scheme
+                    {
+                        String[] photoURLParts = photoURL.split(",");
+                        if (photoURLParts.length > 1) {
+                            photoURL = photoURLParts[1];
```

Comment:
The input parameter `photoURL` must not be assigned to.

Reply:
What exactly do you mean with "must not be assigned to"? From the functional point of view, such an operation is harmless as it doesn't modify the callers data. Do you want to avoid reusing the photoURL because it is a method input argument? The thing is that the photo URL can be both either a payload (e.g. "F3A693C...") or a data URI scheme ("data:image/png;base64,F3A693C..."). By removing the header, the photoURL is still in a format that could also come directly from the user and thus still applies to the "method contract". I would agree if I would misuse the method argument for any other data.
Anyway, I can change it, but then I would introduce a new local variable that is either set to the input argument directly or to the photoURLParts[1]. Is that what you have in mind?

Reply:
Yes please.

Reply:
fixed